src/components/EditCoursePage/index.jsx

```diff
@@ -90,31 +100,48 @@ class EditCoursePage extends React.Component {
       },
     } = this.props;
 
-    // We don't need to send the course runs as part of courseData because they are
-    // pulled out and sent to their own endpoint for saving.
-    delete courseData.course_runs;
-    courseData.subjects = [
-      courseData.subjectPrimary,
-      courseData.subjectSecondary,
-      courseData.subjectTertiary,
-    ].filter(subject => !!subject);
-    courseData.entitlements = entitlements && entitlements[0] && [{
-      mode: courseData.mode,
-      price: courseData.price,
-      sku: entitlements[0].sku,
-    }];
-    courseData.image = courseData.imageSrc;
-    // We are removing imageSrc so we don't send the image data twice
-    delete courseData.imageSrc;
-    courseData.video = { src: courseData.videoSrc };
-    courseData.key = key;
-    courseData.uuid = uuid;
-    /* eslint-enable no-param-reassign */
+    let updatingPublishedRun = false;
+    if (targetRun) {
+      const submittedRun = courseRuns.find(run => run.key === targetRun.key);
+      // If we are updating a published course run, we need to also publish the course.
+      // We want to use the same indicator of draft = false for consistency.
+      if (submittedRun.status === PUBLISHED) {
+        updatingPublishedRun = true;
+      }
+    }
+
+    return {
+      additional_information: courseData.additional_information,
+      draft: !updatingPublishedRun,
+      entitlements: entitlements && entitlements[0] && [{
```

Comment:
I'm actually not sure we need this `entitlements && entitlements[0] &&` check here because we are not creating draft entitlements for Audit seats.